packages/devtools_app/lib/src/ui/flutter/service_extension_widgets.dart

```diff
@@ -390,36 +391,25 @@ mixin _ServiceExtensionMixin<T extends _ServiceExtensionWidget> on State<T> {
     if (disabled) {
       return;
     }
+
     setState(() {
       disabled = true;
     });
-    // TODO(https://github.com/flutter/devtools/issues/1249): Avoid adding
-    // and removing snackbars so often as we do here.
-    ScaffoldFeatureController<SnackBar, SnackBarClosedReason> snackBar;
+
     if (widget.inProgressText != null) {
-      Scaffold.of(context)
-          .removeCurrentSnackBar(reason: SnackBarClosedReason.remove);
-      // Push a snackbar that the action is in progress.
-      snackBar = Scaffold.of(context).showSnackBar(
-        SnackBar(content: Text(widget.inProgressText)),
-      );
+      // TODO(devoncarew): Display this 'starting work' message in the status
+      // bar.
+
     }
+
     try {
       await action();
-      // If the action was successful, remove the snack bar and show a new
-      // one with action success.
-      snackBar?.close();
+
       if (widget.completedText != null) {
-        Scaffold.of(context).showSnackBar(
-          SnackBar(content: Text(widget.completedText)),
-        );
+        Notifications.of(context)?.push(widget.completedText);
```

Comment:
Having this `?.` here makes me nervous that we will miss showing some notifications to the user. Can we add a TODO with a bug to fix it. I think Dave may have some ideas on what is going wrong in this case.

Reply:
Fyi @DaveShuckerow

Reply:
I'll add a TODO:. Note that this was only ever null for tests - for regular app runs this was always non-null.

Reply:
we may have to add a test wrapper that provides Notifications. See how `wrapWithControllers` is used in tests.

Reply:
Thanks! the `wrap()` call looked most similar to what I need. I added a new `warpWithNotifications()` call in this test file.

Reply:
(and removed the use of the null aware operator)